src/Types/Workout.ts

```diff
@@ -0,0 +1,114 @@
+import { Resources } from './Resources';
+export type WorkoutObjectType =
+  | 'exuipment'
+  | 'position'
+  | 'exercise'
+  | 'assignment'
+  | 'set'
+  | 'workout';
+
+export type Reference<T> = string | T;
+
+export interface WorkoutObject {
+  name: string; // unique string used to reference the object
+  label?: string; // string used for display, defaults to name
+  type?: WorkoutObjectType; // if not speified will be inffered
+  resources?: Reference<Resources>; // reference to a media resources object
+  // if not specified default media will be used
+}
+
+// e.g. palnk, down-facing-dog, transition
+// the name Position is not the best for a timed trnsition
+// Position duration can be left unspecified, or specified by trainer
+export interface Position extends WorkoutObject {
+  duration?: number; // Duration is oprional, may be assigned by trainer  // Referende to a positionobject
+}
+// Equipment metric can be left unspecified, or specified by trainer
+export interface Equipment extends WorkoutObject {
+  metric?: [number, number]; // range of weight, band number, etc
+  unit: string; // lbs for weight, band No. etc..
+}
+
+//e.g. : pushup, sun-salutaion,  dumbell-snatch
+// if flow is not specified, exercise will get a defualt duration
+// if equipment not specified assume none
+export interface Exercise extends WorkoutObject {
+  flow?: Reference<Position>[];
+  equipment?: Reference<Equipment>;
+}
+
+// A set of 1 or more exercises that will count as a rep
+export interface Assignment extends WorkoutObject {
+  exercises: Reference<Exercise>[]; // the repeatable unit from now up.
+}
+
+// if duration is not speifed, a default duration will be used
+// an empty Break is useful as it is a reference to media resources
+export interface Break extends WorkoutObject {
+  duration?: number;
+}
+
+//
+export interface Set extends WorkoutObject {
+  assignments: Reference<Assignment>[];
+  metric?: number; // if not specified defaults to 1 for count 30 for duration
+  measure?: 'count' | 'duration'; // if not specified defaults to count
```

Comment:
Let's formally define this:
```
export enum Measurement {
  duration,
  reps,
  breaths,
  weight,
}
```

so that we can say `measurement: Measurement`